pkg/worker/worker.go

```diff
@@ -39,27 +39,30 @@ func init() {
 // 1. Output data will be placed into an agreed upon results directory.
 // 2. The Job will wait for a done file
 // 3. The done file contains a single string of the results to be sent to the master
-func GatherResults(waitfile string, url string, client *http.Client) error {
-	var inputFileName []byte
-	var err error
-	var outfile *os.File
-
-	// just loop looking for a file.
-	logrus.Infof("Waiting on: (%v)", waitfile)
+func GatherResults(waitfile string, url string, client *http.Client, stop <-chan struct{}) error {
+	logrus.WithField("waitfile", waitfile).Info("Waiting for waitfile")
+	ticker := time.Tick(1 * time.Second)
 	for {
-		if inputFileName, err = ioutil.ReadFile(waitfile); err == nil {
-			break
+		select {
+		case <-ticker:
+			if inputFileName, err := ioutil.ReadFile(waitfile); err == nil {
+				return handleWaitFile(string(inputFileName), url, client)
+			}
+		case <-stop:
+			logrus.Info("Forced shutdown. Stopping.")
```

Comment:
Yeah I'd rather see the sleep here than in the handler.

Reply:
ah. ignore that other comment then, that sounds totally reasonable.